src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2229,19 +2229,24 @@ void os::Linux::print_process_memory_info(outputStream* st) {
   // Print glibc outstanding allocations.
   // (note: there is no implementation of mallinfo for muslc)
 #ifdef __GLIBC__
+#if __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 33)
+  struct mallinfo2 mi = ::mallinfo2();
+  const size_t total_allocated = mi.uordblks;
+  const bool might_have_wrapped = false;
+#else // __GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 33)
   struct mallinfo mi = ::mallinfo();
-
   // mallinfo is an old API. Member names mean next to nothing and, beyond that, are int.
   // So values may have wrapped around. Still useful enough to see how much glibc thinks
   // we allocated.
-  const size_t total_allocated = (size_t)(unsigned)mi.uordblks;
-  st->print("C-Heap outstanding allocations: " SIZE_FORMAT "K", total_allocated / K);
+  const size_t total_allocated = static_cast<size_t>(mi.uordblks);
```

Comment:
The original version was correct here, please restore it. You need to make the int value unsigned before extending it to 64bit size_t, otherwise you get sign extension for values >= 2g. 

One could argue that the second cast to size_t in the original coding was superfluous, but the cast to unsigned was needed.